lib/Doctrine/Common/Annotations/DocParser.php

```diff
@@ -1054,30 +1061,20 @@ private function Values(): array
             $token = $this->lexer->lookahead;
             $value = $this->Value();
 
-            if (! is_object($value) && ! is_array($value)) {
-                throw $this->syntaxError('Value', $token);
-            }
-
             $values[] = $value;
         }
 
+        $namedArguments      = [];
+        $positionalArguments = [];
         foreach ($values as $k => $value) {
             if (is_object($value) && $value instanceof stdClass) {
-                $values[$value->name] = $value->value;
-            } elseif (! isset($values['value'])) {
-                $values['value'] = $value;
+                $namedArguments[$value->name] = $value->value;
             } else {
-                if (! is_array($values['value'])) {
-                    $values['value'] = [$values['value']];
-                }
-
-                $values['value'][] = $value;
+                $positionalArguments[$k] = $value;
             }
-
-            unset($values[$k]);
         }
 
-        return $values;
+        return ['named_arguments' => $namedArguments, 'positional_arguments' => $positionalArguments];
```

Comment:
In `resolvePositionalValues`, you validate `$positionalArguments`. Can't we do this here already? I don't like that we pass around potentially invalid metadata.

Reply:
The `Values()` method is part of the parsing process. If we move the validation inside, it will have to know what kind of annotation we are talking about. The validation is different is you use `@NamedArgumentConstructor` or not. 
This is valid without `@Name("1", foo="bar", "2")` but should not be valid with `@NamedArgumentConstructor` as it's supposed to mimic a regular constructor call with named arguments.

Reply:
All right, thank you for clarifying. 👍🏻